phantun/src/bin/client.rs

```diff
@@ -111,12 +119,9 @@ async fn main() -> io::Result<()> {
 
     let ipv4_only = matches.get_flag("ipv4_only");
 
-    let remote_addr = tokio::net::lookup_host(matches.get_one::<String>("remote").unwrap())
-        .await
-        .expect("bad remote address or host")
-        .find(|addr| !ipv4_only || addr.is_ipv4())
-        .expect("unable to resolve remote host name");
-    info!("Remote address is: {}", remote_addr);
+    let ip4p_resolve = matches.get_flag("ip4p_remote");
+
+    let remote_addr = lookup_host(matches.get_one::<String>("remote").unwrap(), ip4p_resolve, ipv4_only).await;
```

Comment:
This will only resolve the DNS once on program start but the use case for IP4P usually means both port and IP is dynamic and would require update from time to time.